src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php

```diff
@@ -43,6 +44,10 @@ protected function describeRouteCollection(RouteCollection $routes, array $optio
 
         $tableRows = array();
         foreach ($routes->all() as $name => $route) {
+            if (!preg_match("/{$filter}/", $name)) {
```

Comment:
IMO we should avoid the call to `preg_match()` if we already know that the `filter` param isn't provided.